cumulusci/core/config/__init__.py

```diff
@@ -11,19 +11,37 @@
 class ConnectedAppOAuthConfig(BaseConfig):
     """Salesforce Connected App OAuth configuration"""
 
-    pass
+    client_id: str
+    client_secret: str
+    login_url: str
+    callback_url: str
 
 
 class FlowConfig(BaseConfig):
     """A flow with its configuration merged"""
 
-    pass
+    description: str
+    steps: list
+    group: str
+    checks: list
+    project_config: BaseConfig
 
 
 from cumulusci.core.config.org_config import OrgConfig
 
 
 class ServiceConfig(BaseConfig):
+    url: str
+    username: str
+    password: str
+    token: str
+    email: str
+    client_id: str
+    client_secret: str
+    token_uri: str
+    callback_url: str
+    login_url: str
```

Comment:
Ooh, I don't like this. The valid service attributes depend on what type of service it is. And new service types with new attributes can be defined in cumulusci.yml

Reply:
@prescod I didn't mean to imply that solving all that is the only acceptable way to make incremental improvements here. But I wanted to point out where I see potential for things to not work out cleanly. And I do feel rather strongly that adding _incorrect_ type annotations would be a step backward. If we can't take the time to get the type annotations correct right now, maybe it would be better to use a different mechanism for specifying what names are expected (like just a set assigned to a class attribute) for the time being.

With regard to point 4 about dynamically generating type annotations, that sounds pretty messy and pretty far from the notion of type annotations as something that can be statically analyzed. Instead of that, it might make sense to move in the direction of making each type of service use its own ServiceConfig subclass that declares its attributes as a pydantic model. That could replace the existing configuration of service attributes in yaml; I'm not sure that has much benefit. Not as part of this PR though!

Reply:
Okay I added an opt-in type checking mode to verify that the types are at least correct enough to pass our test suite. 059de0e8a